src/firefox/index.js

```diff
@@ -193,6 +193,7 @@ export type ConfigureProfileOptions = {
 
 export type ConfigureProfileFn = (
   profile: FirefoxProfile,
+  customs: {},
```

Comment:
it is hard to be completely sure that a custom property will be completely valid/recognized by Firefox (e.g. if you open the "about:config" url in a Firefox tab you can briefly take a look at the one defined in your Firefox profile), what we can check is probably that:

- preference name:
  - it should have at least one "."
  - it should always start with a lower case letter
(actually in the preferences names used in Firefox everything on the left of the first `"."` is always composed only by lowercase letters)
  - it should contain only valid characters (e.g. lowercase and uppercase letters, numbers, and a couple of special characters like `.` , `-`, `_` and `*`)
- preference value: 
  - it can only be integers, booleans or strings (everything that is not a boolean or an integer)

Most of the above validations on the preference names seems to be doable with regular expressions (https://developer.mozilla.org/en/docs/Web/JavaScript/Guide/Regular_Expressions) that are still readable from us human beings :-) 

(we can write some really amazing/crazy things with regex syntaxes but, at least in my own opinion, if it is possible, it is better to use an amount of them that make the final expression still reasonably readable)

On the bright side, if someone really needs this feature, he usually knows pretty well what he is doing, and so the above validations should be more then enough (maybe some of them are even too much).

About the "where it would be better to validate the custom preferences", in general I think that is always better to validate the external inputs only on the "border" of our software systems (which means as near as possible to where the external input has been received) so that we are not going to check them everywhere.

In this case a good position to validate the custom preferences is probably inside the `execute` method of the `Program` class :
https://github.com/mozilla/web-ext/blob/master/src/program.js#L83

(it could be probably nicer if we define and export the helper function that will validate these `customPrefs` object in the `src/firefox/preferences.js` module and then import and call it from the `execute` method in the `src/program.js` module)